apps/portal/src/content/docs/components/file-explorer.mdx

```diff
@@ -0,0 +1,208 @@
+    </div>
+  } value="src" link="/src">
+    src
+  </FileExplorer.FolderItem>
+
+  <FileExplorer.FolderItem content={
+    <div className="flex flex-col">
+      <FileExplorer.FileItem link="/src/index.tsx">index.tsx</FileExplorer.FileItem>
+      <FileExplorer.FileItem link="/src/file-explorer.tsx">FileExplorer.tsx</FileExplorer.FileItem>
+      <FileExplorer.FileItem link="/src/type.ts">type.ts</FileExplorer.FileItem>
+    </div>
+  } value="assets" link="/assets">
+    assets
+  </FileExplorer.FolderItem>
+  <FileExplorer.FileItem link="/package.json">package.json</FileExplorer.FileItem>
+</FileExplorer.Root>`}
+/>
+
+## Usage
+
+```typescript jsx
+import { FileExplorer } from '@harnessio/ui/components'
+
+return (
+  <FileExplorer.Root
+    value={['src']}                         // [REQUIRED] Currently selected items
+    onValueChange={(value) => console.log('Value changed:', value)} // [REQUIRED] Handler for selection changes
+  >
+    <FileExplorer.FolderItem
+      value="src"                          // [OPTIONAL] Folder identifier
+      link="/src"                          // [REQUIRED] Navigation link
+      isActive={false}                     // [OPTIONAL] Active state
+      chevronClassName="custom-chevron"    // [OPTIONAL] Custom chevron styling
+    >
+      src
+      <FileExplorer.FileItem
+        link="/src/index.tsx"              // [OPTIONAL] Navigation link
+        isActive={false}                   // [OPTIONAL] Active state
+      >
+        index.tsx
+      </FileExplorer.FileItem>
+    </FileExplorer.FolderItem>
+    <FileExplorer.FileItem link="/package.json">package.json</FileExplorer.FileItem>
+  </FileExplorer.Root>
+)
+```
+
+## Anatomy
+
+The File Explorer component can be used to display a hierarchical view of the file system and handle file or directory selection.
+
+```typescript jsx
+<FileExplorer.Root
+  value={['src']}                         // [REQUIRED] Currently selected items
+  onValueChange={(value) => console.log('Value changed:', value)} // [REQUIRED] Handler for selection changes
+>
+  <FileExplorer.FolderItem
+    value="src"                          // [OPTIONAL] Folder identifier
+    link="/src"                          // [REQUIRED] Navigation link
+    isActive={false}                     // [OPTIONAL] Active state
+    chevronClassName="custom-chevron"    // [OPTIONAL] Custom chevron styling
+  >
+    src
+    <FileItem
+      link="/src/index.tsx"              // [OPTIONAL] Navigation link
+      isActive={false}                   // [OPTIONAL] Active state
+    >
+      index.tsx
+    </FileExplorer.FileItem>
+  </FileExplorer.FolderItem>
+  <FileExplorer.FileItem link="/package.json">package.json</FileExplorer.FileItem>
+</FileExplorer.Root>
+```
+
+## API Reference
+
+### Root
+
+The `Root` component serves as the main container for displaying the file system hierarchy and handling file or directory selection.
```

Comment:
Add a bare-bones example of this component with all props along with comment descriptions. The [`HoverCard` docs](https://harness-design.netlify.app/components/hover-card/#root) are a decent example of this